platform/ios/src/MGLMapView.mm

```diff
@@ -4765,14 +4765,17 @@ - (void)updateAnnotationViews
             annotationView.center = [self convertCoordinate:annotationContext.annotation.coordinate toPointToView:self];
         }
     }
-    
-    CGPoint upperLeft = {_largestAnnotationViewSize.width,_largestAnnotationViewSize.height};
-    CGPoint lowerRight = {CGRectGetWidth(self.bounds) + _largestAnnotationViewSize.width,
-                          CGRectGetHeight(self.bounds) + _largestAnnotationViewSize.height};
-    
-    CLLocationCoordinate2D upperLeftCoordinate = [self convertPoint:upperLeft toCoordinateFromView:self];
-    CLLocationCoordinate2D lowerRightCoordinate = [self convertPoint:lowerRight toCoordinateFromView:self];
-    
+
+    CGFloat largestHeight = _largestAnnotationViewSize.height;
+    CGFloat largestWidth = _largestAnnotationViewSize.width;
+    CGFloat viewHeight = CGRectGetHeight(self.frame);
+    CGFloat viewWidth = CGRectGetWidth(self.frame);
+    CLLocationCoordinate2D southwestCoordinate = [self convertPoint:{-largestWidth, viewHeight + largestHeight}
+                                               toCoordinateFromView:self];
+    CLLocationCoordinate2D northeastCoordinate = [self convertPoint:{viewWidth + largestWidth, -largestHeight}
+                                               toCoordinateFromView:self];
+    MGLCoordinateBounds coordinateBounds = {southwestCoordinate, northeastCoordinate};
```

Comment:
This code could use `UIEdgeInsetsInsetRect()` to apply a slop area around the map view’s bounds, then convert it using `-convertRect:toCoordinateBoundsFromView:`.